articles/142_idl.md

```diff
@@ -35,3 +35,53 @@ Further it describes how the interfaces are used to generate code in C, C++ and
 <div class="alert alert-warning" markdown="1">
   <b>TODO:</b> All of this too
 </div>
+
+### Type Mapping
+
+The next table defines how IDL types are mapped to the following programming languages:
+
+* C11 (or higher)
+* C++11 (or higher)
+* Python 3 (or higher)
+
+| IDL type    | C type      | C++ type    | Python type          |
+| --------    | ----------- | ----------- | -------------------- |
+| float       | float       | float       | float                |
+| double      | double      | double      | float                |
+| long double | long double | long double | float                |
+| char        | signed char | char        | str with length 1    |
+| wchar       | char16\_t   | char16\_t   | str with length 1    |
+| bool        | \_Bool      | bool        | bool                 |
+| octet       | uint8\_t    | uint8\_t    | bytes with length 1  |
+| int8        | int8\_t     | int8\_t     | int                  |
+| uint8       | uint8\_t    | uint8\_t    | int                  |
+| int16       | int16\_t    | int16\_t    | int                  |
+| uint16      | uint16\_t   | uint16\_t   | int                  |
+| int32       | int32\_t    | int32\_t    | int                  |
+| uint32      | uint32\_t   | uint32\_t   | int                  |
+| int64       | int64\_t    | int64\_t    | int                  |
+| uint64      | uint64\_t   | uint64\_t   | int                  |
+
+The next table defines how IDL templated and constructed types are mapped to the programming languages (unless specified otherwise below).
+
+| IDL type       | C type                 | C++ type            | Python type |
+| -------------- | ---------------------- | ------------------- | ----------- |
+| T[N]           | T[N]                   | std::array<T, N>    | list        |
+| sequence<T>    | struct {size\_t, T * } | std::vector<T>      | list        |
+| sequence<T, N> | struct {size\_t, T * } | custom\_class<T, N> | list        |
```

Comment:
should this be `struct {size\_t, T * , N }` instead? (where N would be the max capacity)

Reply:
I think max should to be a `const` variable; because if it's non-const then it would need to be in two places. First in the struct that can be modified by a user, and second in code used to validate a message.

If that sounds reasonable then it implies the struct with a field for max capacity would be generated for every field. Alternatively a struct definition could be reused if the max capacity was a separate const variable.

```c
const size_t example_msgs__example_bounded__sensor_readings__max_capacity = 64;
//  ...
struct example_msgs__example_bounded
{
  // ...
  rcutils_float_array * sensor_readings = NULL;
}
```